arduino-ide-extension/src/browser/dialogs/settings/settings.ts

```diff
@@ -137,10 +136,6 @@ export class SettingsService {
       ['en', ...(await this.localizationProvider.getAvailableLanguages())],
       this.localizationProvider.getCurrentLanguage(),
       this.preferenceService.get<number>(FONT_SIZE_SETTING, 12),
-      this.preferenceService.get<string>(
-        'workbench.colorTheme',
-        'arduino-theme'
```

Comment:
What about generalizing [this](https://github.com/arduino/arduino-ide/blob/5087ff08f216773c35de7ecc8300b771eafea4a9/electron/build/patch/frontend/index.js#L17-L22) logic, and replacing the hard-coded `'arduino-theme'` default value with it?

Something like this:
```ts
this.preferenceService.get<string>(
  'workbench.colorTheme',
  window.matchMedia &&
    window.matchMedia('(prefers-color-scheme: dark)').matches
    ? 'arduino-theme-dark'
    : 'arduino-theme'
),
```

Reply:
For me it seemed logical to use the `ThemeService` as we go on to use `ThemeService.get` in the JSX of the `settings-component.tsx`.

I'm probably missing some info. too... on my mac, on first start with a dark themed os, the Theia (Dark) theme is loaded not the Arduino one, is this unexpected?

Are you suggesting wrapping that logic? or just reusing it as above?

Reply:
> For me it seemed logical to use 

The method is `loadSettings`; but it does not load the theme. It uses the value from the `ThemeService`. All other settings are read from the `settings.json` or the `arduino-cli.yaml`. Your implementation relies on how the `workbench.colorTheme` preference value changes are picked up by the `ThemeService` form Theia, but it can change. I am proposing this change to have complete control of our code.

> Are you suggesting wrapping this logic? or just reusing it as above?

Reusing the same logic would be the best, but since the theme hack will be removed, you can reuse the code here.

Reply:
understood, lets go for that then